packages/server/modules/workspaces/repositories/users.ts

```diff
@@ -12,3 +20,89 @@ export const setUserActiveWorkspaceFactory =
       meta.set(userId, 'isProjectsActive', isProjectsActive)
     ])
   }
+
+const buildInvitableCollaboratorsByProjectIdQueryFactory =
+  ({ db }: { db: Knex }) =>
+  ({
+    workspaceId,
+    projectId,
+    search
+  }: {
+    workspaceId: string
+    projectId: string
+    search?: string
+  }) => {
+    const query = tables
+      .users(db)
+      .join(WorkspaceAcl.name, WorkspaceAcl.col.userId, Users.col.id)
+      .join(Streams.name, Streams.col.workspaceId, WorkspaceAcl.col.workspaceId)
+      .where(WorkspaceAcl.col.workspaceId, workspaceId)
+      .whereNotIn(
+        Users.col.id,
+        tables
+          .streamAcl(db)
+          .select(StreamAcl.col.resourceId)
+          .whereNot(StreamAcl.col.resourceId, projectId)
+      )
+    if (search) {
+      query
+        .join(UserEmails.name, UserEmails.col.userId, Users.col.id)
+        .andWhere((w) =>
+          w
+            .whereLike(Users.col.name, `%${search}%`)
+            .orWhereLike(UserEmails.col.email, `%${search}%`)
+        )
+    }
+    return query.groupBy(Users.col.id)
+  }
+
+export const getInvitableCollaboratorsByProjectIdFactory =
+  ({ db }: { db: Knex }) =>
+  async ({
+    filter,
+    cursor,
+    limit
+  }: {
+    filter: {
+      workspaceId: string
+      projectId: string
+      search?: string
+    }
+    cursor?: string
+    limit: number
+  }): Promise<WorkspaceTeamMember[]> => {
+    const { workspaceId, projectId, search } = filter
+    const query = buildInvitableCollaboratorsByProjectIdQueryFactory({ db })({
+      workspaceId,
+      projectId,
+      search
+    })
+    if (cursor) {
+      query.andWhere(Users.col.createdAt, '<', cursor)
```

Comment:
ideally we should use decodeCompositeCursor/encodeCompositeCursor and order by both a date and also user id, to ensure we don't run into issues when multiple users have the same date